### PR TITLE
Make linting include subcharts and use strict mode

### PR DESCRIPTION
**Pull request by ashrayjain**

### CODE REVIEW

**Review by tabboud — DISMISSED** [review]

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -24,15 +24,6 @@ jobs:
       - name: Set up chart-testing
         uses: helm/chart-testing-action@v2.6.0
 
-      - name: Run chart-testing (list-changed)
-        id: list-changed
-        run: |
-          changed=$(ct list-changed --config .github/linters/ct.yaml)
-          if [[ -n "$changed" ]]; then
-            echo "changed=true" >> "$GITHUB_OUTPUT"
```
Comment (ashrayjain) [inline .github/workflows/lint.yaml:32]: the ct tool already checks for changed helm charts so this is a bit redundant

File: .github/linters/ct.yaml [hunk 2]
```diff
@@ -3,12 +3,12 @@ remote: origin
 target-branch: develop
 debug: true
 validate-maintainers: false
+helm-lint-extra-args: '--strict --with-subcharts'
 chart-dirs:
   - charts/alpha
   - charts/beta
   - charts/ga
 chart-repos:
-  - bitnami=https://charts.bitnami.com/bitnami
```
Comment (ashrayjain) [inline .github/linters/ct.yaml:11]: since this is declared as oci, we dont need it

**Review by tabboud — CHANGES_REQUESTED** [review]
Code changes look good but there are a few README updates we should make now
1. The `Packaging` and following example should include a step to run `helm dependency build <chart-path>` before running the package step.
2. The chart READMEs still contain references to the local sub-charts README. These should be updated to the upstream URL instead.

File: README.md [hunk 3]
```diff
@@ -37,8 +37,10 @@ This repository uses chart specific labels to tag versions of each chart. The la
 Each helm chart can be packaged separately and published to a container registry.
 
 ```shell
+# Build helm dependencies for a single helm-chart product
+$ helm dependency build ./charts/<helm-chart-path> --repository-config helm-repositories.yaml
 # Package a single helm-chart product
-helm package -d ./build ./charts/<helm-chart-name>
+helm package -d ./build ./charts/<helm-chart-path>
```
Comment (tabboud) [inline README.md]: nit: Lets add the `$` to this and the command below since we have some inconsistency in how they are written

**Review by tabboud — COMMENTED** [review]
Still need to update the chart README's to no longer reference the in-repo values file for the upstream charts. Otherwise this looks good to me.

File: README.md (hunk 3 above)
Comment (ashrayjain) [reply README.md]: fixed

**Review by tabboud — APPROVED** [review]

**Pull request merged**